### PR TITLE
Exclude specific tags

### PR DESCRIPTION
**Pull request by itkg-yfarci**
add process to exclude specific elements from HTTP request

### CODE REVIEW

File: HttpCache/BaseStore.php [hunk 1]
```diff
@@ -11,20 +11,23 @@ class BaseStore implements StoreInterface
     protected $root;
     protected $keyCache;
     protected $locks;
+    protected $tagsToExclude;
```
Comment (clement-michelet) [inline HttpCache/BaseStore.php]: phpDoc, initialize with array

File: HttpCache/BaseStore.php [hunk 2]
```diff
@@ -11,20 +11,23 @@ class BaseStore implements StoreInterface
...
 
     /**
      * Constructor.
      *
      * @param string $root The path to the cache directory
+     * @param array $tagsToExclude   Removes the specified members from the set value stored at key
      */
-    public function __construct($root)
+    public function __construct($root, $tagsToExclude)
```
Comment (clement-michelet) [inline HttpCache/BaseStore.php]: typehint array, initialize with array

File: HttpCache/BaseStore.php [hunk 3]
```diff
@@ -397,7 +400,15 @@ protected function getCacheKey(Request $request)
      */
     protected function persistRequest(Request $request)
     {
-        return $request->headers->all();
+        $header = $request->headers->all();
+
+        if ($this->tagsToExclude) {
```
Comment (clement-michelet) [inline HttpCache/BaseStore.php]: With array typehint on the parameter, that line could be removed.

File: HttpCache/BaseStore.php [hunk 4]
```diff
@@ -397,7 +400,15 @@ protected function getCacheKey(Request $request)
...
+            foreach($this->tagsToExclude as $tag) {
```
Comment (clement-michelet) [inline HttpCache/BaseStore.php]: Not PSR-2 compliant: http://www.php-fig.org/psr/psr-2/#5-5-foreach

File: HttpCache/HttpCache.php [hunk 5]
```diff
@@ -21,6 +21,14 @@ protected function createStore()
         if (false === $redisConnection = $this->envOptions['redis_connection']) {
             throw new \RuntimeException("Please define redis_connection parameter for the Itkg\\CombinedHttpCache.");
         }
+        
+        $tagsToExclude = array();
+        if (isset($this->envOptions['redis_tags_exclude']) && !is_array($this->envOptions['redis_tags_exclude'])){
+            throw new \RuntimeException("Please define redis_tags_exclude parameter for the Itkg\\CombinedHttpCache with an array.");
+        }elseif(isset($this->envOptions['redis_tags_exclude'])){
```
Comment (clement-michelet) [inline HttpCache/HttpCache.php]: Not PSR-2 compliant: http://www.php-fig.org/psr/psr-2/#5-1-if-elseif-else

File: HttpCache/HttpCache.php [hunk 6]
```diff
@@ -21,6 +21,14 @@ protected function createStore()
         if (false === $redisConnection = $this->envOptions['redis_connection']) {
             throw new \RuntimeException("Please define redis_connection parameter for the Itkg\\CombinedHttpCache.");
         }
+        
+        $tagsToExclude = array();
```
Comment (clement-michelet) [inline HttpCache/HttpCache.php:25]: I don't see any use of that variable.

File: HttpCache/Store.php [hunk 7]
```diff
@@ -17,12 +17,13 @@ class Store extends BaseStore
     /**
      * @param string $cacheConnectionDsn    Normalized connection params for the cache client
      * @param string $root                  The path to the cache directory
+     * @param array $tagsToExclude          Removes the specified members from the set value stored at key
      */
-    public function __construct($cacheConnectionDsn, $root)
+    public function __construct($cacheConnectionDsn, $root, $tagsToExclude)
```
Comment (clement-michelet) [inline HttpCache/Store.php]: typehint array + initialize array

**Comment by pdenis** [comment]
Ne pas inclure de nom de projet dans un bundle tiers

File: HttpCache/BaseStore.php (hunk 1 above)
Comment (itkg-yfarci) [reply HttpCache/BaseStore.php]: done

File: HttpCache/BaseStore.php (hunk 2 above)
Comment (itkg-yfarci) [reply HttpCache/BaseStore.php]: done

File: HttpCache/BaseStore.php (hunk 3 above)
Comment (itkg-yfarci) [reply HttpCache/BaseStore.php]: done

File: HttpCache/BaseStore.php (hunk 4 above)
Comment (itkg-yfarci) [reply HttpCache/BaseStore.php]: done

File: HttpCache/HttpCache.php (hunk 5 above)
Comment (itkg-yfarci) [reply HttpCache/HttpCache.php]: done

File: HttpCache/HttpCache.php (hunk 6 above)
Comment (itkg-yfarci) [reply HttpCache/HttpCache.php:25]: done

File: HttpCache/Store.php (hunk 7 above)
Comment (itkg-yfarci) [reply HttpCache/Store.php]: done

**Comment by itkg-yfarci** [comment]
J'ai supprimé le nom du projet du commentaire

File: HttpCache/BaseStore.php [hunk 8]
```diff
@@ -13,18 +13,25 @@ class BaseStore implements StoreInterface
...
+    protected $tagsToExclude = array();
+
+    /**
      * Constructor.
      *
      * @param string $root The path to the cache directory
+     * @param array $tagsToExclude   Removes the specified members from the set value stored at key
      */
-    public function __construct($root)
+    public function __construct($root, array $tagsToExclude)
```
Comment (clement-michelet) [inline HttpCache/BaseStore.php]: Replace by `array $tagsToExclude = array()` to make it optional

File: HttpCache/Store.php [hunk 9]
```diff
@@ -17,12 +17,13 @@ class Store extends BaseStore
     /**
      * @param string $cacheConnectionDsn    Normalized connection params for the cache client
      * @param string $root                  The path to the cache directory
+     * @param array $tagsToExclude          Removes the specified members from the set value stored at key
      */
-    public function __construct($cacheConnectionDsn, $root)
+    public function __construct($cacheConnectionDsn, $root, array $tagsToExclude)
```
Comment (clement-michelet) [inline HttpCache/Store.php]: Replace by `array $tagsToExclude = array()` to make it optional

File: HttpCache/BaseStore.php (hunk 8 above)
Comment (itkg-yfarci) [reply HttpCache/BaseStore.php]: done

File: HttpCache/Store.php (hunk 9 above)
Comment (itkg-yfarci) [reply HttpCache/Store.php]: done

**Comment by clement-michelet** [comment]
:+1:

**Comment by Vowow** [comment]
:+1:

**Pull request merged**